### PR TITLE
Add generic button

### PR DESCRIPTION
**Pull request by itkg-nanne**

### CODE REVIEW

File: BackofficeBundle/Controller/ContentController.php [hunk 1]
```diff
@@ -42,7 +42,8 @@ public function formAction(Request $request, $contentId)
                     'contentId' => $content->getContentId(),
                     'language' => $content->getLanguage(),
                     'version' => $content->getVersion(),
-                ))
+                )),
+                'delete_button' => (!$content->isUsed())
             ), ContributionActionInterface::EDIT);
```
Comment (alavieille) [inline BackofficeBundle/Controller/ContentController.php]: add parameter  ContributionActionInterface::EDIT is this uselless

File: BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Service/Form/Mixin/FormViewButtonsMixin.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    /**
+     * Redirect to new workflow profile view
+     *
+     * @param {mixed}  data
+     * @private
+     */
+    _redirectNewElement(data) {
+        let message = new FlashMessage(data, 'success');
+        FlashMessageBag.addMessageFlash(message);
```
Comment (alavieille) [inline BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Service/Form/Mixin/FormViewButtonsMixin.js]: all controller don't add message in response on server, so this method don't work for all form

File: BackofficeBundle/Resources/views/Form/form_div_layout.html.twig [hunk 3]
```diff
@@ -681,22 +678,10 @@
...
-            {% set label = 'open_orchestra_backoffice.form.delete' %}
-            {% set id = 'delete_' ~ form.vars.name %}
-            {% set button_class= 'danger' %}
-            {{ block('button_widget') }}
-        {% endif %}
-    {% endspaceless %}
-{% endblock oo_site_submit_widget %}
-
-{% block oo_block_submit_widget %}
+{% block generic_submit_buttons %}
```
Comment (alavieille) [inline BackofficeBundle/Resources/views/Form/form_div_layout.html.twig]: why no refacto directly form_submit_widget ?

**Comment by itkg-nanne** [comment]
update done @alavieille

File: BackofficeBundle/Resources/views/Form/form_div_layout.html.twig [hunk 4]
```diff
@@ -678,36 +690,18 @@
...
-            {% set attr = {'class': 'delete-button'} %}
-            {% set button_class= 'danger' %}
-            {{ block('button_widget') }}
-        {% endif %}
-    {% endspaceless %}
-{% endblock generic_submit_buttons %}
-
 {% block oo_site_submit_widget %}
-    {{ block('generic_submit_buttons') }}
+    {{ block('form_submit_widget') }}
```
Comment (alavieille) [inline BackofficeBundle/Resources/views/Form/form_div_layout.html.twig]: this block is useless now by default form_submit_widget is call same for other

File: ApiBundle/Controller/ContentTypeController.php [hunk 5]
```diff
@@ -138,6 +138,9 @@ public function deleteAction($contentTypeId)
     {
         if (0 == $this->get('open_orchestra_model.repository.content')->countByContentType($contentTypeId)) {
             $contentTypes = $this->get('open_orchestra_model.repository.content_type')->findBy(array('contentTypeId' => $contentTypeId));
+            foreach ($contentTypes as $contentType) {
```
Comment (alavieille) [inline ApiBundle/Controller/ContentTypeController.php]: if you can delete a version you can delete all no ?

File: GroupBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Group/GroupFormView.js [hunk 6]
```diff
@@ -11,6 +11,7 @@ class GroupFormView extends mix(AbstractFormView).with(FormViewButtonsMixin)
     /**
      * Initialize
      * @param {Form}   form
+     * @param {String} statusId
```
Comment (alavieille) [inline GroupBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Group/GroupFormView.js]: ?? groupId no ?

File: WorkflowAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/Router/Status/StatusRouter.js [hunk 7]
```diff
@@ -39,11 +42,15 @@ class StatusRouter extends AbstractWorkflowRouter
      *
      * @param  {String} statusId
      */
-    editStatus(statusId) {
+    editStatus(statusId, name) {
```
Comment (alavieille) [inline WorkflowAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/Router/Status/StatusRouter.js]: js doc

File: WorkflowAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/Router/WorkflowProfile/WorkflowProfileRouter.js [hunk 8]
```diff
@@ -76,7 +78,11 @@ class WorkflowProfileRouter extends AbstractWorkflowRouter
         let url = Routing.generate('open_orchestra_workflow_admin_workflow_profile_form', {workflowProfileId: workflowProfileId});
         this._displayLoader(Application.getRegion('content'));
         FormBuilder.createFormFromUrl(url, (form) => {
-            let workflowProfileFormView = new WorkflowProfileFormView({form: form});
+            let workflowProfileFormView = new WorkflowProfileFormView({
+                form: form,
+                workflowProfileId: workflowProfileId,
+                name: name
```
Comment (alavieille) [inline WorkflowAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/Router/WorkflowProfile/WorkflowProfileRouter.js]: ?? name is not defined in this method

File: WorkflowAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Status/StatusFormView.js [hunk 9]
```diff
@@ -1,31 +1,75 @@
...
  * @class StatusFormView
  */
-class StatusFormView extends AbstractFormView
+class StatusFormView extends mix(AbstractFormView).with(FormViewButtonsMixin)
 {
     /**
+     * Initialize
+     * @param {Form}   form
+     * @param {String} statusId
+     * @param {Array}  name
```
Comment (alavieille) [inline WorkflowAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Status/StatusFormView.js]: array ?

File: WorkflowAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/WorkflowProfile/WorkflowProfileFormView.js [hunk 10]
```diff
@@ -1,15 +1,32 @@
...
  * @class WorkflowProfileFormView
  */
-class WorkflowProfileFormView extends AbstractFormView
+class WorkflowProfileFormView extends mix(AbstractFormView).with(FormViewButtonsMixin)
 {
     /**
+     * Initialize
+     * @param {Form}   form
+     * @param {String} workflowProfileId
+     * @param {Array}  name
```
Comment (alavieille) [inline WorkflowAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/WorkflowProfile/WorkflowProfileFormView.js]: array ? and what is the name when is the new ?

File: WorkflowAdminBundle/Resources/public/template/Status/statusFormView._tpl.html [hunk 11]
```diff
@@ -3,5 +3,5 @@
         <%= Translator.trans('open_orchestra_backoffice.back_to_list') %>
     </a>
 </div>
-<h1> <%= Translator.trans('open_orchestra_workflow_admin.status.title_edit') %></h1>
+<h1> <%= name %></h1>
```
Comment (alavieille) [inline WorkflowAdminBundle/Resources/public/template/Status/statusFormView._tpl.html]: remove the translation in interface.yml

File: WorkflowAdminBundle/Resources/public/template/WorkflowProfile/workflowProfileFormView._tpl.html [hunk 12]
```diff
@@ -3,5 +3,5 @@
         <%= Translator.trans('open_orchestra_backoffice.back_to_list') %>
     </a>
 </div>
-<h1> <%= Translator.trans('open_orchestra_workflow_admin.workflow_profile.title_edit') %></h1>
```
Comment (alavieille) [inline WorkflowAdminBundle/Resources/public/template/WorkflowProfile/workflowProfileFormView._tpl.html]: remove the translation in interface.yml

File: WorkflowAdminBundle/Resources/public/template/Status/newStatusView._tpl.html [hunk 13]
```diff
@@ -1,7 +0,0 @@
-<div class="top-action-buttons">
-    <a class="btn btn-border btn-return" href="#<%= Backbone.history.generateUrl('listStatus') %>" role="button">
-        <%= Translator.trans('open_orchestra_backoffice.back_to_list') %>
-    </a>
-</div>
-<h1> <%= Translator.trans('open_orchestra_workflow_admin.status.title_new') %></h1>
```
Comment (alavieille) [inline WorkflowAdminBundle/Resources/public/template/Status/newStatusView._tpl.html]: remove the translation in interface.yml

File: ApiBundle/Controller/ContentTypeController.php (hunk 5 above)
Comment (itkg-nanne) [inline ApiBundle/Controller/ContentTypeController.php]: That's what is done. The repository send all version of contentType as the selection is done on contentTypeId and not id.

File: WorkflowAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/WorkflowProfile/WorkflowProfileFormView.js (hunk 10 above)
Comment (itkg-nanne) [inline WorkflowAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/WorkflowProfile/WorkflowProfileFormView.js]: name is Translator.trans('open_orchestra_workflow_admin.workflow_profile.title_new') (line 66 in WorkflowProfileRouter)

File: WorkflowAdminBundle/Resources/public/template/Status/newStatusView._tpl.html (hunk 13 above)
Comment (itkg-nanne) [inline WorkflowAdminBundle/Resources/public/template/Status/newStatusView._tpl.html]: I can't. Used by router

**Comment by itkg-nanne** [comment]
update done @alavieille

**Comment by alavieille** [comment]
:+1:

**Pull request merged**